editor/src/main/com/mbrlabs/mundus/ui/modules/Outline.java

```diff
@@ -470,26 +481,32 @@ public void show(GameObject go, float x, float y) {
             if (selectedGO != null) {
                 //Activate menu options for selected game objects
                 rename.setDisabled(false);
-                //duplicate.setDisabled(false);
                 delete.setDisabled(false);
+
             } else {
                 //disable MenuItems which only work with selected Item
                 rename.setDisabled(true);
-                //duplicate.setDisabled(true);
                 delete.setDisabled(true);
             }
+            //terrain can not be duplicated
+            if (selectedGO == null || selectedGO.findComponentByType(Component.Type.TERRAIN) != null) {
+                duplicate.setDisabled(true);
+            } else {
+                duplicate.setDisabled(false);
+            }
+
         }
 
         public void showRenameDialog() {
             final Tree.Node node = tree.findNode(selectedGO);
-            final GameObject go = (GameObject) node.getObject();
+            final TreeNode goNode = (TreeNode) node.getActor();
 
             InputDialog renameDialog = Dialogs.showInputDialog(Ui.getInstance(), "Rename", "", new InputDialogAdapter() {
                 @Override
                 public void finished(String input) {
-                    Log.traceTag(TAG, "Rename game object [{}] to [{}].", go, input);
+                    Log.traceTag(TAG, "Rename game object [{}] to [{}].", goNode.go, input);
                     //update name
-                    go.name = input;
+                    goNode.setLabel(input);
```

Comment:
Yes it is irritating. But i have to update the game object in the sceneGraph AND in the outline. So i need a method to rename both.
I will rename the setLabel method into rename, if thats better. The problem is, the name is not stored in a central object, because the sceneGraph objects have no interface to the VisUI Objects. Thats the real problem when updating the outline ...